src/index/ExternalSortFunctors.h

```diff
@@ -65,4 +66,26 @@ struct SortText {
   }
 };
 
+// A comparator that sorts rows by a runtime-specified list of column indices.
+// Uses simple `<` comparison on Ids (internal order).
+struct SortByColumns {
+  std::vector<ColumnIndex> sortColumns_;
+
+  explicit SortByColumns(std::vector<ColumnIndex> sortColumns)
+      : sortColumns_{std::move(sortColumns)} {}
+
+  // Default constructor for template requirements.
+  SortByColumns() = default;
+
+  template <typename T1, typename T2>
+  bool operator()(const T1& a, const T2& b) const {
```

Comment:
Off-Topic; This is a scenario where JIT can shine:
If the JIT detects that the sort columns are always a small vector it can generate machine code that unrolls this loop for all the common sizes. Achieving the same in C++ is possible, but very cumbersome (it's basically what we do whenever `ad_utility:callFixedSize` is being used).